docs/modular-resources/create/_index.md

```diff
@@ -34,45 +34,92 @@ Before you start coding a new resource model, follow these steps to find the app
+2. [Code a main program](#code-a-main-entry-point-program), <file>main.py</file>, that starts the module after adding your desired resources from the registry.
    Import your custom model and API into the main program and register them with your chosen SDK.
    This main program is the "entry point" to your module.
 
 3. [Compile or package](#compile-the-module-into-an-executable) the module into a single executable that can receive a socket argument from Viam, open the socket, and start the module at the entry point.
 
-### Code a new resource model
+{{% /tab %}}
+{{% tab name="Go" %}}
 
-The following example module registers a modular resource implementing Viam's built-in [Base API](/components/base/#api) [(rdk:service:base)](/modular-resources/key-concepts/#models) as a new model, `"mybase"`, using the model family `acme:demo:mybase`.
+1. [Code a new resource model](#code-a-new-resource-model) server. It must have all the methods that the Viam RDK requires, and should match the built-in API client {{< glossary_tooltip term_id="subtype" text="subtype" >}} like ['rdk:component:base'](https://pkg.go.dev/go.viam.com/rdk@v0.10.0/components/base#pkg-functions).
+   Save this in a file within your module’s <file>my_modular_resource.go</file> file.
+   Name your model according to the namespace of the built-in API you are implementing.
+   For more information see [Naming your model](/modular-resources/key-concepts/#naming-your-model-namespacerepo-namename).
+2. [Code a main program](#code-a-main-entry-point-program), <file>main.go</file>, that starts the module after adding your desired resources from the registry.
+   Import your custom model and API into the main program and register them with your chosen SDK.
+   This main program is the "entry point" to your module.
 
-- For more information, see [Naming your model](/modular-resources/key-concepts/#naming-your-model-namespacerepo-namename).
+3. [Compile or package](#compile-the-module-into-an-executable) the module into a single executable that can receive a socket argument from Viam, open the socket, and start the module at the entry point.
 
-The Go code for the custom model (<file>mybase.go</file>) and module entry point file (<file>main.go</file>) is adapted from the full demo modules available on the [Viam GitHub](https://github.com/viamrobotics/rdk/blob/main/examples/customresources).
+{{% /tab %}}
+{{< /tabs >}}
+
+### Code a new resource model
+
+The following example module registers a modular resource implementing Viam's built-in [Base API](/components/base/#api) [(rdk:service:base)](/modular-resources/key-concepts/#models) as a new model, `"mybase"`, using the model family `acme:demo:mybase`.
+For more information, see [Naming your model](/modular-resources/key-concepts/#naming-your-model-namespacerepo-namename).
```

Comment:
Ah, this is why we've used `base` throughout. Unless we change this example also to `arm`, let's keep `base` above, but just add some clarification to indicate that we're talking about the component type, not the base level of functionality or the basic featureset, etc.